h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -590,16 +590,18 @@ public void cv_mainModelScores(int N, ModelMetrics.MetricBuilder mbs[], ModelBui
 
     // Compute and put the cross-validation metrics into the main model
     Log.info("Computing "+N+"-fold cross-validation metrics.");
-    mainModel._output._cross_validation_models = new Key[N];
+    Key<M>[] cvModKeys = new Key[N];
+    mainModel._output._cross_validation_models = _parms._keep_cross_validation_models ? cvModKeys : null;
```

Comment:
I would move this to the end of the changes. It is better to keep the _output field null to prevent accidental changes.

Reply:
not sure I understand what you mean here: with the code above, the output field stays null if the property is disabled, following same pattern as with predictions below.

Reply:
If you move line 594 to lets say 636 or later you will ensure nobody will access `mainModel._output._cross_validation_models` accidentally. This code will just be a little bit safer and prevents (artificial) scenarios like this

    mainModel._output._cross_validation_models = mainModel._output._cross_validation_models.clone();
    modify(mainModel._output._cross_validation_models);
    // code assumes cvModKeys and mainModel._output._cross_validation_models are the same but they are not

This is not real issue right now. Up to you.

Reply:
TBH mainModel is accessible in the entire method so I don't think moving lines around has a big impact, so I'll leave those lines where they were from the beginning.
The rest is psychology of reading :) some people could argue that by reading the initialization on the model output property early in the method, they'll be more aware that they shouldn't change it later... I think the difference is too tiny to deserve having a strong opinion.

Reply:
closed